fri/src/prover.rs

```diff
@@ -104,13 +110,28 @@ where
         }
     }
 
-    // We should be left with `blowup` evaluations of a constant polynomial.
-    assert_eq!(folded.len(), config.blowup());
-    let final_poly = folded[0];
-    for x in folded {
-        assert_eq!(x, final_poly);
+    // Previously, we checked that folded was constant at this point. Now, we transform folded
+    // from the evaluation domain to the coefficient domain and verify the trailing coefficients.
+    let mut final_poly = folded.clone();
+
+    // Switch from the evaluation basis to the coefficient basis.
+    reverse_slice_index_bits(&mut final_poly);
+    final_poly = Radix2Dit::default().idft(final_poly);
```

Comment:
This is almost certainly not a bottleneck but, we could do this faster by running `Radix2Dit` on only the first half of `final_poly`. (Or less than this depending on `log_blowup`). Maybe add a comment or add this computation to the debug timing output? (using `debug_span!`)

Reply:
Added a TODO.